winml/lib/Api.Image/EventTimer.h

```diff
@@ -0,0 +1,29 @@
+#include "pch.h"
+
+class EventTimer
+{
+public:
+  EventTimer()
+  {
+  }
+  
+  bool Start()
+  {
+    auto now = std::chrono::high_resolution_clock::now();
+    if (!_started || 
+         std::chrono::duration_cast<std::chrono::microseconds>(now - _startTime).count() > _kDurationBetweenSendingEvents)
+    {
+      _started = true;
+      _startTime = std::chrono::high_resolution_clock::now();
```

Comment:
Should it be thread safe?

Reply:
well, all the usage of the class is inside locks, so I didn't want to put more locks in here. But I suppose if someone wanted to use it in a place that didn't already have locks that might be problematic. What do you think?

Reply:
It's fine as is